assembly/src/errors.rs

```diff
@@ -324,6 +324,28 @@ impl AssemblyError {
     }
 }
 
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub struct SerializationError {
+    message: String,
+}
```

Comment:
Instead of using a struct with a single message, we should probably use an enum with different variants. For example:
```Rust
pub enum SerializationError {
    InvalidOpcode,
    InvalidBoolValue,
}
```